src/connection.js

```diff
@@ -0,0 +1,112 @@
+const { Map, fromJS } = require('immutable')
+const OpSet = require('./op_set')
+
+// Returns true if all components of clock1 are less than or equal to those of clock2.
+// Returns false if the clocks are incomparable or if clock1 is greater than clock2.
```

Comment:
When would a clock be incomparable? Reading the code I don't see an obvious case when that would be.

Reply:
Two clocks are "incomparable" if one of them is greater in some fields, and the other is greater in other fields. That is, the clocks represent two states that have diverged.

Reply:
Ah, I think that language is a bit imprecise. They are comparable, it's just that neither is less than or equal to the other. When I read "incomparable", I was looking for degenerate data or things like that.

Reply:
It's a standard term in mathematics (since vector clocks encode a [partial order](https://en.wikipedia.org/wiki/Partially_ordered_set)). But anyway, I've (hopefully) clarified the comment in aaee2ad.